### PR TITLE
WIP: Chore: support new myst-nb

### PR DESCRIPTION
**Pull request by agoose77**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,7 +1,83 @@
 [build-system]
-requires = ["setuptools>=46.4.0", "wheel"]
-build-backend = "setuptools.build_meta"
+requires = [
+    "hatchling>=1.3.1",
+]
+build-backend = "hatchling.build"
```
Comment (chrisjsewell) [inline pyproject.toml:5]: hmm, would rather ebp just standardised around flit, which is already now used by most packages, unless there is a particular reason not to

Comment (agoose77) [reply pyproject.toml:5]: Sure, I prefer hatchling but I hadn't appreciated that we're a mainly flit workshop at this stage.

Comment (agoose77) [reply pyproject.toml:5]: @chrisjsewell can you move this to #98

Comment (ofek) [reply pyproject.toml:5]: > unless there is a particular reason

Hey, Hatch maintainer here! One reason is for plugins, so you could do things like versioning using just Git or compiling your codebase with Mypyc to speed things up.

Also, Project Jupyter is almost done [switching](https://github.com/jupyterlab/team-compass/issues/146) their entire packaging ecosystem. I tried to link to most of the changes here: https://hatch.pypa.io/latest/users/

Comment (agoose77) [reply pyproject.toml:5]: Yes, personally I'm a big fan of Hatch over Flit. I think Flit is a great tool, but Hatch is more customisable / extensible in my experience.

The nice thing is that with PEP 631 (thanks @ofek) we can switch fairly easily between the two.

**Comment by agoose77** [comment]
Closed by #104

**Comment by mmcky** [comment]
hey @agoose77 thanks for starting this effort. Sorry I didn't see this PR until now. @AakashGfude has been working on supporting the new myst-nb (as you would have seen). Didn't mean to step on your toes. I'll check the open PR's more closely and try and liaise with you next time.

**Comment by AakashGfude** [comment]
Hey sorry @agoose77, missed this one. Will note to look more at PRs, before starting.

**Pull request closed without merging**